### PR TITLE
Jazzy patch release versions  2024-07-01

### PR DESCRIPTION
**Pull request by marcoag**
First Jazzy patch release :tada: 

CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21348/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15662/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22090/)
* RHEL [Build Status](https://ci.ros2.org/job/ci_linux-rhel/866/)

Packaging:
* Linux [Build Status](http://ci.ros2.org/job/ci_packaging_linux/559/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_packaging_linux-aarch64/107/)
* Windows [Build Status](http://ci.ros2.org/job/ci_packaging_windows/228/)
* RHEL [Build Status](https://ci.ros2.org/job/ci_packaging_linux-rhel/65/)

### CODE REVIEW

File: ros2.repos [hunk 1]
```diff
@@ -2,15 +2,15 @@ repositories:
   ament/ament_cmake:
     type: git
     url: https://github.com/ament/ament_cmake.git
-    version: 2.5.0
+    version: 2.5.1
```
Comment (clalancette) [inline ros2.repos:5]: There was a late-breaking fix that you might want to get into this patch release: https://github.com/ament/ament_cmake/pull/539 .

**Review by clalancette — CHANGES_REQUESTED** [review]
Besides what I have left inline, you might also want to consider updating cyclonedds to 0.10.5, which will include all of the latest fixes.  If you do that, don't forget to also do a `bloom-release` for it.

Finally, keep a close eye on the CI for this.  I know that Jazzy still has a number of warnings, but look out for failing tests.  If you see a bunch of failing tests related to `rmw_fastrtps`, we may have to update Fast-RTPS to 2.14.1, like we did for Rolling.

**Comment by marcoag** [comment]
Re-running CI with `ament_cmake`  `2.5.2`:

CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21351/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15664/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22093/)
* RHEL [Build Status](https://ci.ros2.org/job/ci_linux-rhel/868/)

File: ros2.repos (hunk 1 above)
Comment (marcoag) [reply ros2.repos]: Thanks for the heads up here, released: https://github.com/ros/rosdistro/pull/41915.

**Comment by marcoag** [comment]
Since there's suspicious results from the buildfarm I'll run the last release and the current patch release so we can compare and identify errors.

CI `jazzy-release` branch :
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21357/)

CI `jazzy_patch_release_1` branch:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21358/)

**Comment by marcoag** [comment]
Ok since the `jazzy_patch_release_` job hanged running a new one with a bump on fastrtps to `2.14.1`

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21360/)

**Comment by clalancette** [comment]
> Ok, since the `jazzy_patch_release_1` job hanged running a new one with a bump on `Fast-DDS` to `2.14.1`

All right, it looks like that one passed, and with the same warnings as the `jazzy-release` branch.

So I think we need to release 2.14.1 into Jazzy.  Note that this *isn't* the latest tag on https://github.com/eProsima/Fast-DDS, so you'll have to manually specify it when doing the `bloom-release` for it.

Separately, it might be a good idea to release cyclonedds 0.10.5 into Jazzy as well.  That should fix most of the failing tests on Linux.

**Comment by marcoag** [comment]
CI with `cyclone`  `0.10.5`:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21363/)

**Comment by marcoag** [comment]
The rest of CI:

* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15671/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22100/)
* RHEL [Build Status](https://ci.ros2.org/job/ci_linux-rhel/874/)

**Review by clalancette — APPROVED** [review]
Assuming CI is green-ish (including the packaging jobs), this looks good to me!

**Comment by marcoag** [comment]
Packaging:
* Linux [Build Status](http://ci.ros2.org/job/ci_packaging_linux/560/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_packaging_linux-aarch64/108/)
* Windows [Build Status](http://ci.ros2.org/job/ci_packaging_windows/229/)
* RHEL [Build Status](https://ci.ros2.org/job/ci_packaging_linux-rhel/66/)

**Comment by marcoag** [comment]
And seem to be green-ish, with some shadows of yellow. I'll take it as good and go ahead with this delayed sync! :cry:

**Comment by marcoag** [comment]
* Windows [Build Status](http://ci.ros2.org/job/ci_packaging_windows/230/)

**Pull request merged**